syntax = "proto3";

package api.v1;

import "google/protobuf/empty.proto";
import "telempb/telem.proto";

service RangeCreateService {
    rpc Exec(RangeCreateRequest) returns (RangeCreateResponse) {}
}

service RangeRetrieveService {
    rpc Exec(RangeRetrieveRequest) returns (RangeRetrieveResponse) {}
}

service RangeKVSetService {
    rpc Exec(RangeKVSetRequest) returns (google.protobuf.Empty) {}
}

service RangeKVGetService {
    rpc Exec(RangeKVGetRequest) returns (RangeKVGetResponse) {}
}

service RangeKVDeleteService {
    rpc Exec(RangeKVDeleteRequest) returns (google.protobuf.Empty) {}
}

message Range {
    string key = 1;
    telempb.TimeRange time_range = 2;
    string name = 3;
}

message RangeCreateRequest {
    repeated Range ranges = 1;
}

message RangeCreateResponse {
    repeated Range ranges = 1;
}

message RangeRetrieveRequest {
    repeated string keys = 1;
    repeated string names = 2;
}

message RangeRetrieveResponse {
    repeated Range ranges = 1;
}

message RangeKVSetRequest {
    string range_key = 1;
    map<string, string> pairs = 2;
}

message RangeKVGetRequest {
    string range_key = 1;
    repeated string keys = 2;
}

message RangeKVGetResponse {
    map<string, string> pairs = 1;
}

message RangeKVDeleteRequest {
    string range_key = 1;
    repeated string keys = 2;
}

